### PR TITLE
 Fix the size of all-null blocks returned from selective readers

### PR DESCRIPTION
**Pull request by bhhari**
When all data in a column is null, `SelectiveStreamReader.getBlock()` may return a block with more positions when requested. If this column happens to be inside of a struct and there are other non-null columns, RowBlock constructor will throw an exception: `IllegalArgumentException: length of field blocks differ`

```
@Test
public void testAllNulls()
{
    getQueryRunner().execute("CREATE TABLE test_all_nulls AS SELECT orderkey, cast(row(null, 1) AS ROW(null_boolean BOOLEAN, a INTEGER)) as struct FROM orders");

    try {
        assertQuery("SELECT * FROM test_all_nulls WHERE struct IS NOT NULL AND orderkey % 2 = 0", "SELECT orderkey, (null, 1) FROM orders WHERE orderkey % 2 = 0");
    }
    finally {
        getQueryRunner().execute("DROP TABLE test_all_nulls");
    }
}
```

This test fails with an exception in RowBlock: “length of field blocks differ”
 
```
Caused by: java.lang.IllegalArgumentException: length of field blocks differ: field 0: 2, block 1: 1
         at com.facebook.presto.spi.block.RowBlock.validateConstructorArguments(RowBlock.java:93)
         at com.facebook.presto.spi.block.RowBlock.fromFieldBlocks(RowBlock.java:52)
         at com.facebook.presto.orc.reader.StructSelectiveStreamReader.getBlock(StructSelectiveStreamReader.java:427)
         at com.facebook.presto.orc.OrcSelectiveRecordReader.getNextPage(OrcSelectiveRecordReader.java:393)
         at com.facebook.presto.hive.orc.OrcSelectivePageSource.getNextPage(OrcSelectivePageSource.java:84)
         ... 12 more
```

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by mbasmanova** [comment]
@bhhari Here is what a test might look like:

```
    @Test
    public void testAllNullsInStruct()
    {
        List<String> types = ImmutableList.of(BOOLEAN, TINYINT, SMALLINT, INTEGER, BIGINT, DOUBLE, REAL, VARCHAR, DATE);

        String query = String.format("SELECT orderkey, CAST(ROW(%s, 1) AS ROW(%s, a INTEGER)) as struct FROM orders",
                types.stream()
                        .map(type -> "null")
                        .collect(joining(", ")),
                types.stream()
                        .map(type -> String.format("null_%s %s", type.toLowerCase(), type.toUpperCase()))
                        .collect(joining(", ")));

        getQueryRunner().execute("CREATE TABLE test_all_nulls_in_struct AS " + query);

        try {
            for (String type : types) {
                assertQuery(format("SELECT struct.a, struct.null_%s FROM test_all_nulls_in_struct WHERE struct IS NOT NULL AND orderkey %% 2 = 0", type.toLowerCase()),
                        "SELECT 1, null FROM orders WHERE orderkey % 2 = 0");
            }
        }
        finally {
            getQueryRunner().execute("DROP TABLE test_all_nulls_in_struct");
        }
    }
```

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java [hunk 1]
```diff
@@ -666,11 +706,11 @@ public void testPushdownComposition()
         // Tests composing two pushdowns each with a range filter and filter function.
         assertQuery(
                 "WITH data AS (" +
-                    "    SELECT l.suppkey, l.linenumber, l.shipmode, MAX(o.orderdate)" +
-                    "    FROM lineitem l,  orders o WHERE" +
-                    "        o.orderkey = l.orderkey AND linenumber IN (2, 3, 4, 6) AND shipmode LIKE '%AIR%'" +
-                    "        GROUP BY l.suppkey, l.linenumber, l.shipmode)" +
-                    "SELECT COUNT(*) FROM data WHERE suppkey BETWEEN 10 AND 30 AND shipmode LIKE '%REG%'");
+                        "    SELECT l.suppkey, l.linenumber, l.shipmode, MAX(o.orderdate)" +
```
Comment (bhhari) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java]: there is a bad formatting on the exiting file. When auto format is used, this gets changed. This should fix this once for all, but sorry for the unrelated change here
